src/operations.cpp

```diff
@@ -1564,7 +1566,9 @@ void create_symlink(const path& to, const path& from, error_code* ec)
 BOOST_FILESYSTEM_DECL
 path current_path(error_code* ec)
 {
-# ifdef BOOST_POSIX_API
+# if defined(__wasm)
+  return ".";
```

Comment:
This is not right, `current_path` must return an absolute path (see C++20 [fs.op.current.path]). For example, `copy` implementation relies on that, and probably not only that.

If the platform does not support `current_path`, it should fail with `BOOST_ERROR_NOT_SUPPORTED` (`ENOSYS`), but then I'm not sure how useful the rest of the library would be as it is used in implementation of quite a few operations.

Reply:
As an alternative, you could assume that the current path is always the root directory. I'm not sure how valid is that in WASI (i.e. is `equivalent(".", "/") == true`).

Reply:
Also, see if there is a way to obtain path from a file descriptor or from the result of `stat` on WASI.

Reply:
Understood. I'll check whether this assumption holds (the precise WASI semantics around `.` are not entirely clear to me) and see what can be done.

Reply:
> Also, see if there is a way to obtain path from a file descriptor or from the result of `stat` on WASI.

That's what the [`fd_prestat_dir_name` syscall](https://github.com/WebAssembly/WASI/blob/master/phases/snapshot/docs.md#-fd_prestat_dir_namefd-fd-path-pointeru8-path_len-size---errno) is for.

Reply:
Can you use it to obtain the full path to '.'?

Reply:
> If the platform does not support `current_path`, it should fail with `BOOST_ERROR_NOT_SUPPORTED` (`ENOSYS`), but then I'm not sure how useful the rest of the library would be as it is used in implementation of quite a few operations.

I've changed the code to do that. It turns out that the application I'm porting is only using `boost::filesystem::path`, so it's still useful for that. Failing with `BOOST_ERROR_NOT_SUPPORTED` seems to be a safe choice that can be always revisited later, perhaps with input from WASI folks.

Reply:
> Can you use it to obtain the full path to '.'?

No. There is no concept of an "absolute path" in WASI, since it eliminates ambient authority and uses capabilities for everything; the `/<something>` directory, `.` directory, and `..` directory behave effectively like separate mounts, akin to the way `.` and `..` work on FAT, if you've seen that. It's kind of weird.